VinylUI.Tests/WinFormsTests.fs

```diff
@@ -410,3 +427,59 @@ let ``getObjConverter for record option type handles nulls`` () =
     let converter = BindPartExtensions.getObjConverter<Book option> ()
     converter.ToSource null |> shouldEqual None
     converter.ToControl None |> shouldEqual null
+
+type ListControls = ListType | ComboType
+
+let listControls = [ ListType; ComboType ]
+
+[<TestCaseSource("listControls")>]
+let ``Model to view correctly updates SelectedIndex to -1`` controlType =
+    use form = new FakeForm()
+    let ctrl =
+        match controlType with
+        | ComboType -> form.ComboBox :> ListControl
+        | ListType -> form.ListBox :> ListControl
+
+    Bind.model(<@ model.Books @>).toDataSource(ctrl, <@ fun b -> b.Id, b.Name @>) |> ignore
+    let getList () = ctrl.DataSource :?> Book seq |> Seq.toList
+    getList () |> shouldEqual model.Books
+
+    let viewExpr = <@ ctrl.SelectedIndex @>
+    let binding = Bind.view(viewExpr).toModel(<@ model.BookIndex @>)
+    binding |> testModelToView viewExpr model.BookIndex 1 1
+    binding |> testModelToView viewExpr model.BookIndex -1 -1
+
+[<Test>]
+let ``Model to view correctly updates SelectedItem to null`` () =
+    use form = new FakeForm()
+    let ctrl = form.ComboBox
+
+    Bind.model(<@ model.BookObjs @>).toDataSource(ctrl, <@ fun b -> b.Id, b.Name @>) |> ignore
+    let getList () = ctrl.DataSource :?> BookObj seq |> Seq.toList
+    getList () |> shouldEqual model.BookObjs
+
+    let viewExpr = <@ ctrl.SelectedItem @>
+    let binding = Bind.view(viewExpr).toModel(<@ model.BookSelection @>)
+
+    ctrl.SelectedIndex <- 1
```

Comment:
this should not be necessary since the binding is supposed to set the view to the model's value immediately on bind. If this is necessary for the test, further investigation is warranted.